eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqlitePaymentsDb.scala

```diff
@@ -33,14 +33,36 @@ class SqlitePaymentsDb(sqlite: Connection) extends PaymentsDb with Logging {
   import SqliteUtils.ExtendedResultSet._
 
   val DB_NAME = "payments"
-  val CURRENT_VERSION = 2
+  val CURRENT_VERSION = 3
 
-  using(sqlite.createStatement()) { statement =>
-    require(getVersion(statement, DB_NAME, CURRENT_VERSION) <= CURRENT_VERSION, s"incompatible version of $DB_NAME DB found") // version 2 is "backward compatible" in the sense that it uses separate tables from version 1. There is no migration though
+  // 1 -> 2 was backwards compatible so this method is noop
+  def migration12(statement: Statement): Unit = ()
+
+  def migration23(statement: Statement): Unit = {
```

Comment:
That doesn't seem right: if we are in version 2 then we already have those tables.

That should be:

```scala
def migration23(statement: Statement): Unit = {
    statement.executeUpdate("CREATE INDEX IF NOT EXISTS payment_hash_idx ON sent_payments(payment_hash)")
  }

Right?
```

Reply:
Also the original comment for version 2 was much more informative

Reply:
All `CREATE TABLE/INDEX` has `IF NOT EXISTS` which neutralizes them.

Reply:
Sure, that's how the `case CURRENT_VERSION =>` (because starting from nothing or being up to date is the same from a migration point of view) but there is no need to use that in the `2->3` scenario.